src/main/java/Utils/Appointment.java

```diff
@@ -0,0 +1,239 @@
+            Connection connection = DriverManager.getConnection("jdbc:derby://localhost:1527/Appointments", username, password);
+            PreparedStatement prep = connection.prepareStatement(sql);
+
+            LocalTime s = LocalTime.parse(event.getStartTime());
+            LocalTime e = LocalTime.parse(event.getEndTime());
+
+            Time start = Time.valueOf(s);
+            Time end = Time.valueOf(e);
+
+            java.sql.Date date = java.sql.Date.valueOf(event.getDate());
+
+            int id = getAppointmentID();
+            int docID = doctors.get(event.getDoctorName());
+            String docName = event.getDoctorName();
+            String custName = event.getCustomerName();
+            String type = event.getType();
+
+            prep.setDate(1, date);
+            prep.setInt(2, id);
+            prep.setTime(3, start);
+            prep.setInt(4, docID);
+            prep.setTime(5, end);
+            prep.setString(6, custName);
+            prep.setString(7, type);
+            prep.setString(8, docName);
+            prep.executeUpdate();
+
+        } catch (ClassNotFoundException | SQLException e) {
+            System.out.println(e);
+        }
+    }
+
+    public int getAppointmentID() {
+        int count = 0;
+        try {
+            Class.forName("org.apache.derby.jdbc.ClientDriver");
+            String sql = "SELECT count(*) FROM APPOINTMENTS";
+            Connection connection = DriverManager.getConnection("jdbc:derby://localhost:1527/Appointments", username, password);
+            Statement stmt = connection.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            rs.next();
+            count = rs.getInt(1);
+        } catch (ClassNotFoundException | SQLException e) {
+            System.out.println(e);
+        }
+        return count + 1;
+    }
+
+    public ArrayList<Event> sortAppointmentsByDate(String appointmentDate, ArrayList<Event> event) {
+        ArrayList<Event> e = new ArrayList<>();
+
+        String[] temp = appointmentDate.split("-");
+
+        event.stream().map(new Function<Event, Event>() {
+            @Override
+            public Event apply(Event event1) {
+                System.out.println(event1.getDate());
+                return event1;
+            }
+        }).filter((event1) -> (event1.getDate().equals(appointmentDate))).forEachOrdered((event1) -> {
+            e.add(event1);
+        });
+        return e;
+    }
+
+    public int getDoctorIDbyName(String name) {
+        int docID = doctors.get(name);
+        return docID;
+    }
+
+    public ArrayList<String> getGeneratedSlots(int apptmLength, int doctorID, String date) {
+        ArrayList<Event> currEvents = new ArrayList<>();
+        ArrayList<String> intervals = new ArrayList<>();
+        String[] hours = new String[2];
+        boolean addSlot = false;
```

Comment:
`addSlot` does not need to be given a value here because it will always be set on line `194`